gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -178,6 +215,96 @@ public boolean accept(File file) {
   }
 
   /**
+   * Recursively load job configuration files under given URI of **directory of config files folder**
+   * todo
+   */
+  private static void loadGenericJobConfigsRecursive(List<Properties> jobConfigs, Properties rootProps,
+      Set<String> jobConfigFileExtensions, URI uri)
+      throws ConfigurationException, IOException {
+
+    /* create the generic file system */
+    org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
+    FileSystem filesystem = FileSystem.get(uri, conf);
+
+    Path configDirPath = new Path(uri);
+    if (!filesystem.exists(configDirPath)) {
+      throw new RuntimeException("No specified .properties directory found :" + configDirPath.toString());
+    }
+
+    FileStatus[] propertiesFilesStatus = filesystem.listStatus(configDirPath, PROPERTIES_PATH_FILTER);
+    ArrayList<String> propertiesFiles = new ArrayList<>();
+    for (FileStatus file : propertiesFilesStatus) {
+      propertiesFiles.add(file.getPath().getName());
+    }
+    if (propertiesFiles.size() > 0) {
+      // There should be a single properties file in each directory (or sub directory)
+      if (propertiesFiles.size() != 1) {
+        throw new RuntimeException("Found more than one .properties file in directory: " + configDirPath);
+      }
+
+      // Load the properties, which may overwrite the same properties defined in the parent or ancestor directories.
+      // Open the inputStream, construct a reader and send to the loader for constructing propertiesConfiguration.
+      PropertiesConfiguration propertiesConfiguration = new PropertiesConfiguration();
+      Path uniqueConfigFilePath = configDirPath.suffix(propertiesFiles.get(0));
+      InputStreamReader inputStreamReader = new InputStreamReader(filesystem.open(uniqueConfigFilePath), Charsets.UTF_8);
+      propertiesConfiguration.load(inputStreamReader);
+
+      rootProps.putAll(ConfigurationConverter.getProperties(propertiesConfiguration));
+      inputStreamReader.close();
+    }
+
+    // Get all non-properties files
+    FileStatus[] nonPropFiles = filesystem.listStatus(configDirPath, NON_PROPERTIES_PATH_FILTER);
+    ArrayList<String> names = new ArrayList<>();
+    for (FileStatus nonPropFile : nonPropFiles) {
+      names.add(nonPropFile.getPath().getName());
+    }
+    if (names.size() == 0 ) {
+      return;
+    }
+
+    for (String name : names) {
```

Comment:
This should be `for (FileStatus fileStatus : nonPropFiles)` instead, avoiding an extra call to `getFileStatus`.